### DISCUSSION TITLE
Wildcard for filenames?

### DISCUSSION THREAD

**Discussion by HerroBert** [borgbase/vorta#1261] [General]
Hi!

How to use wildcards for filesnames?
e. g. /home/user/mydoc*

**Comment by real-yfprojects** [comment]
Where? Do you refer to the exclude TextEdit in the `Sources` Tab?

**Reply by HerroBert** [reply]
I meant to select files for backup.

**Reply by real-yfprojects** [reply]
No, but you can specify a folder to backup and exclude all files that do not match `/home/user/mydoc*`.

**Reply by HerroBert** [reply]
so I have to exclude 1034 folders? really? that can't be true...

**Reply by real-yfprojects** [reply]
You can use a *re*-style exclude pattern which vorta's GUI doesn't recognise yet. It will still be passed to vorta's backend which executes the backups.
```regexp
re:/home/user/(?!mydoc).*
```

**Reply by HerroBert** [reply]
I dont want to exclude. I want to include!

**Reply by real-yfprojects** [reply]
If you simply refuse the solution presented to you, I am not able to help you.

**Reply by HerroBert** [reply]
That is no solution. Vorta just cut off borg configuration options. Pure borg can backup foobar*.

**Reply by m3nu** [reply]
We only support files and folders to include, so people can use a file selector. If your use case is different, you may want to keep using Borg via the command line or suggest a pull request if the change benefits a wider range of our users.

**Reply by real-yfprojects** [reply]
> That is no solution. 

What I proposed should work as expected. You include the parent directory and exclude all the files you didn't want in the first place.

> Vorta just cut off borg configuration options. Pure borg can backup foobar*.

This actually isn't a borg feature but a *bash* feature. Anyways I do think that vorta would benefit from a more user-friendly way to achieve this.

**Reply by HerroBert** [reply]
Do you have an idea how many files/folders someone has to exclude of a 10 years old userhome? Millions!
It's like to go into a store and tell everyone what stuff you do not want to buy. 
Any whay to passthrough  "sh:" to borg?

**Reply by real-yfprojects** [reply]
> Do you have an idea how many files/folders someone has to exclude of a 10 years old userhome? Millions! 

That's what regular expressions like the one I already posted in this discussion are for. They are one-liners that on their own are already able to match (exclude in this case) more than billions of files at once.
[Wikipedia on regular expressions](https://en.wikipedia.org/wiki/Regular_expression)
The [Python Regular Expression syntax](https://docs.python.org/3.10/library/re.html?highlight=re#module-re) is the one understood by *borg* (vorta's backend).

> It's like to go into a store and tell everyone what stuff you do not want to buy.

"All the products that aren't Vegemite."

> Any whay to passthrough "sh:" to borg?

Only for excludes as *m3nu* said.
